### DISCUSSION TITLE
[@types/oojs-ui] Missing some jQuery properties

### DISCUSSION THREAD

**Discussion by jwbth** [DefinitelyTyped/DefinitelyTyped#69557] [Issues with a @types package]
This is rather an issue of the OOUI documentation, but these props are still desirable to have.

This is the list of jQuery properties available in a class extending OO.ui.ProcessDialog:
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/33615628/f25c33b1-0f9b-479c-8856-6a4dbdd794d0)

As you can see, `$body`, among others, in not in the list. `this.$body` is used in the example code in the [docs for OO.ui.ProcessDialog](https://doc.wikimedia.org/oojs-ui/v0.44/js/#!/api/OO.ui.ProcessDialog), as well as the [docs for OO.ui.Dialog](https://doc.wikimedia.org/oojs-ui/v0.44/js/#!/api/OO.ui.Dialog). But it is not in the auto-generated property list of the docs.

The properties themselves are defined in OO.ui.Window.prototype.initialize, see the [code](https://doc.wikimedia.org/oojs-ui/v0.44/js/source/Window.html#OO-ui-Window-method-initialize).

[a comment by an automated account is not shown]

**Comment by diskdance** [comment]
This is intentionally hidden because interface can only define public members. Allowing these will require to written the package in `class`, which I didn't choose because of OOUI's LSP violation issue and its ES5 style.

**Reply by jwbth** [reply]
That's OK, but then my request will be to use interfaces, not types, so that they could be extended instead of patching the definitions. In this particular case, I added `$body` to `OO.ui.Window`, because `OO.ui.Window.Props` is an interface. But that won't work with cases when a type is used. E.g. I currently use a patched version of ProcessDialog.d.ts where I do this:
![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/33615628/ca5cb4e3-74a6-4da2-b243-29b08247ac43)
